.github/workflows/release.yml

```diff
@@ -140,23 +138,29 @@ jobs:
       - uses: Swatinem/rust-cache@v2.7.5
       - uses: taiki-e/install-action@v2
         with:
-          tool: git-cliff
+          tool: cargo-set-version
 
       - name: version
         id: get_version
         run: |
           if [[ "${{ github.event_name }}" == "push" && "${{ startsWith(github.ref, 'refs/tags/v') }}" == "true" ]]; then
-            echo "VERSION=${GITHUB_REF#refs/tags/}" >> $GITHUB_OUTPUT
+            # Remove the 'v' prefix for Cargo.toml versions
+            VERSION=${GITHUB_REF#refs/tags/v}
+            echo "VERSION=v$VERSION" >> $GITHUB_OUTPUT
+            cargo set-version $VERSION
           else
+            # For non-tag builds, use 0.0.0-{git-sha} format
+            GIT_SHA=$(git rev-parse --short HEAD)
             echo "VERSION=pr-${{ github.event.pull_request.number }}" >> $GITHUB_OUTPUT
+            cargo set-version "0.0.0-${GIT_SHA}"
           fi
 
       - name: changelog
         id: changelog
         run: |
-          VERSION=$(echo "${{ steps.get_version.outputs.VERSION }}" | sed 's/^v//')
-          # Generate changelog for this specific version
-          git cliff --current --strip header > version_changelog.txt
+          LATEST_TAG=$(git describe --tags --abbrev=0)
+          PREVIOUS_TAG=$(git describe --tags --abbrev=0 --exclude=$LATEST_TAG)
+          gh api repos/fossas/circe/compare/$PREVIOUS_TAG...$LATEST_TAG --jq '.commits[] | select(.author.login != "dependabot[bot]") | .commit.message | select(. | contains("chore") | not) | "- " + .' > version_changelog.txt
```

Comment:
If I'm reading this right then `chore` text in a commit message means the commit will not be added to the changelog. Is that correct? 

I see it's mentioned a bit in conventional commits, could you share that with the team when you feel ready? It was a new resource for me.